src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.Unix.cs

```diff
@@ -62,73 +62,37 @@ internal static unsafe void GetAnsiStringBytes(ReadOnlySpan<char> chars, Span<by
             bytes[actualByteLength] = 0;
         }
 
-        public static IntPtr AllocHGlobal(IntPtr cb)
+        public static unsafe IntPtr AllocHGlobal(IntPtr cb)
         {
-            nuint cbNative = (nuint)(nint)cb;
-
-            // Avoid undefined malloc behavior by always allocating at least one byte
-            IntPtr pNewMem = Interop.Sys.MemAlloc((cbNative != 0) ? cbNative : 1);
-            if (pNewMem == IntPtr.Zero)
-            {
-                throw new OutOfMemoryException();
-            }
-
-            return pNewMem;
+            return (nint)NativeMemory.Alloc((nuint)(nint)cb);
```

Comment:
Are there other libraries that are using Marshal.AllocHGlobal directly that could/should be replaced by direct usage of NativeMemory.Alloc, e.g. where the fact that it's using LocalAlloc isn't necessary?

Reply:
Likely, but I didn't do an overly extensive search for this right now. It seems like a good low hanging fruit to file an issue for.

Reply:
HttpListener uses Marshal.AllocHGlobal for alignment since <https://github.com/dotnet/corefx/pull/25563>: <https://github.com/dotnet/runtime/blob/ec3e0b276b159ab24946f17f2b89d00862488a6a/src/libraries/System.Net.HttpListener/src/System/Net/Windows/RequestContextBase.cs#L122>

(I think the `_originalBlobAddress` arithmetic could be removed because the native-memory buffer cannot be moved by GC.)

Reply:
> It seems like a good low hanging fruit to file an issue for.

A separate issue/PR is fine,  but I would like for us to do the audit/exercise to help confirm the design and benefit from the new APIs wherever we can. Thanks.

Reply:
Logged https://github.com/dotnet/runtime/issues/54297 and called out examples of places where this is not valid to change (notably APIs like `Marshal.StringToPtrHGlobal` on Windows).